extensions/pagespeed/src/form.tsx

```diff
@@ -0,0 +1,67 @@
+import { Form, ActionPanel, Action, useNavigation, Toast, showToast } from "@raycast/api";
+import { useState } from "react";
+import { isDomain, fetchPageSpeed } from "./utils";
+import Main from "./pagespeed";
+
+export default function PagespeedForm() {
+  const { push } = useNavigation();
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [domainError, setDomainError] = useState<string | undefined>();
+
+  const resetErrors = () => {
+    setDomainError(undefined);
+  };
+
+  const checkDomain = (value: string | undefined) => {
+    if (value && value.length == 0) {
+      setDomainError("The domain should't be empty!");
+    } else if (!isDomain(value)) {
+      setDomainError("The domain is invalid!");
+    } else {
+      setDomainError(undefined);
+      return true;
+    }
+    return false;
+  };
+
+  const handleSubmit = async (values: { nameField: string; domainField: string }) => {
+    const check = checkDomain(values.domainField);
+
+    if (!check) {
+      return;
+    }
+    try {
+      setIsLoading(true);
+      await fetchPageSpeed(values.domainField);
+      setIsLoading(false);
+      push(<Main />);
```

Comment:
Could we maybe use [LaunchCommand](https://developers.raycast.com/api-reference/command#launchcommand) instead, so when I hit escape I go to Raycast root instead of going back to the form, did we ever talk abou that?

Reply:
Hey, just fixed this one according to your suggestion.